pkg/node/address.go

```diff
@@ -46,6 +47,7 @@ var (
 	ipv6NodePortAddrs map[string]net.IP // iface name => ip addr
 	ipv4AllocRange    *cidr.CIDR
 	ipv6AllocRange    *cidr.CIDR
+	routerEniInfo     *linuxrouting.RoutingInfo // Applicable in the ENI IPAM mode.
```

Comment:
I wonder if we're better off simply adding the only the fields from the `RoutingInfo` struct that we need, rather than the whole struct. Mostly because it doesn't feel right for this package to be imported inside `pkg/node`.

For example, we can store:

- IPv4 Router CIDRs
- Interface number for ENI device
- MAC addr for ENI device

Reply:
If the main concern is about package dependencies, then we could just define a routing info interface in this package and dependency-inject the routing info. That could be useful for unit-testing the new functionality (eg, make sure that the routing info is used in the right way in this package).

In practice, the above fields represent 3/5 fields in the `RoutingInfo` which is mostly pretty generic-looking, so I'm not sure how much finessing is required here; independently setting various fields which are all derived from the same source of truth doesn't seem like it will bring much benefit (rather it'll just be additional boilerplate).

Reply:
Yeah, from a code readability point of view, it's not clear at all why the node package all of a sudden has a dependency on a thing that is only used in Azure or ENI mode. Consequently, having a variable name called `routerENIInfo`, which is the only reference to anything IPAM-specific in the node package, just seems like we're missing a piece of abstraction. Same issue with the heath endpoint in ENI mode with the daemon package as well.

This feels like something that should be addressed in a later refactor, but for now, I think going with an interface is a decent enough middle-ground.

Reply:
Also the name `linuxrouting` seems to be a bad choice in hindsight. Yes it concerns itself with configuring Linux-specific routes and such, but it's only applicable for ENI and Azure. Naming is hard :/

Reply:
>In practice, the above fields represent 3/5 fields in the RoutingInfo which is mostly pretty generic-looking, so I'm not sure how much finessing is required here

Correct. Also, even though I'm not familiar with AKS, it seems like the `gatewayIP` field is relevant, and could come into play in future.

re: package dependency

I 100% agree that the dependency seems out of place. Sure, interface based abstraction seems reasonable. Thanks!